src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_field_from_props.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { useMemo, useEffect } from 'react';
+import { UseFieldProps } from '../components';
+import { FieldHook, FieldConfig, FieldValidationData } from '../types';
+import { useFormContext } from '../form_context';
+import { useField, InternalFieldConfig } from './use_field';
+
+/**
+ * Initially the code in this hook was inside the <UseField /> component.
+ * It serves to initialize a FieldHook (useField() from "hooks/use_field.ts") based on the props
+ * passsed to the component.
+ * When refactoring the <UseMultiFields /> component I realized that it would be much simpler
```

Comment:
Thanks for clarifying, but this should be a PR comment, not something that lives inside the docs for the component.

Reply:
That's a valid point, I will add it to the PR 👍